### PR TITLE
248312: Implement Azure Kubernetes network policies with Calico

### PR DESCRIPTION
**Pull request by aadev1**
# **What this PR does / why we need it**:
This PR introduces Calico Network policies which help reduce risk of security vulnerabilities, unauthorized access, and a lack of control over network communication within the AKS cluster.

A Global Deny all ingress and egress policy has been added to the Cluster.
Then additional Allow policies have been added only were required for the Cluster and Applications to function properly.

[AB#248312](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/248312)

# Testing
Tested against SND1 ensuring Cluster and Applications are still functioning as normal after Network policies have been applied.  We also did additional testing by spinning up pods in application namespaces and ensuring pod to pod communication within the cluster is allowed and communication from one application namespace to another isn't allowed.

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [ ] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: core/calico/api-server-release.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: release-calico-api-server
+  namespace: flux-config
+spec:
+  chart:
+    spec:
+      chart: adp-calico-helm-chart
+      version: "0.0.1"
```
Comment (DaRoszko) [inline core/calico/api-server-release.yaml:10]: sem-ver is 0.1.0

Comment (aadev1) [reply core/calico/api-server-release.yaml:10]: @DaRoszko PR Created to update helm version to 0.1.0.  https://github.com/DEFRA/adp-core-helm-charts/pull/2.  I'll update this once that is approved.

Comment (aadev1) [reply core/calico/api-server-release.yaml]: @DaRoszko - Updated to 0.1.0

**Review by ajaysoni-cap — DISMISSED** [review]

**Review by DaRoszko — APPROVED** [review]

**Review by ajaysoni-cap — APPROVED** [review]

**Review by ppotturi — APPROVED** [review]

**Review by ganeshg2491 — APPROVED** [review]

File: services/base/repository.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ metadata:
 spec:
   interval: 5m
   ref:
-    branch: main
+    branch: aa/calicoNetworkPolicies
```
[the remark this answers is not in the document]
Comment (aadev1) [reply services/base/repository.yaml]: Switch back to main now.  Thanks @ppotturi

**Pull request merged**